lib/src/main/java/graphql/nadel/validation/NadelTypeValidation.kt

```diff
@@ -159,16 +155,14 @@ internal class NadelTypeValidation {
             errors.add(MissingUnderlyingType(service, overallType))
         }
 
-        return engineSchema
-            .typeMap
-            .asSequence()
-            .filter { (key) ->
-                key in namesUsed
+        return namesUsed
+            .map {
+                engineSchema.typeMap[it]!!
```

Comment:
Does the same thing, but iterates much faster.

Old method:

For each type in the schema (very large set).
Filter out the ones this service used (constant time).

New method:

For each type name the service uses (smaller set).
Look up the type (constant time).